pkg/command/plugin.go

```diff
@@ -58,6 +59,15 @@ func newPluginCmd() *cobra.Command {
 
 	listPluginCmd.Flags().StringVarP(&outputFormat, "output", "o", "", "Output format (yaml|json|table)")
 	listPluginCmd.Flags().StringVarP(&local, "local", "l", "", "path to local plugin source")
+	if config.IsFeatureActivated(constants.FeatureCentralRepository) {
+		// TODO(khouzam): should we completely remove this flag before 1.0, or do we prefer this
+		// more user friendly approach of hiding the flag and then failing in the Run function if it is used?
+		if err := listPluginCmd.Flags().MarkHidden("local"); err != nil {
```

Comment:
I prefer the approach since it provides better explanation to what is happening.
Suggestion: in the future, we can use the code comment to explain the rationale if appropriate, and use github comment threads to discuss the merit of the change.

Reply:
Ok, I'll leave the flag hidden and update the comment.
Next time I'll put a comment in GH instead of in the code when I want to discuss a point 👍